src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,82 @@
+package duke.task;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * It is a base class for specific task types like Todo, Deadline, and Event.
+ */
+public class Task {
+    protected String description; // Changed to private for encapsulation
+    protected boolean isDone; // Changed to private for encapsulation
+
+    /**
+     * Constructs a Task object with a description and initializes it as not done.
+     *
+     * @param description A description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the task as done by setting the completion status to true.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as done by setting the completion status to false.
+     */
+    public void unmarkAsDone() {
```

Comment:
"unmarkAsDone" might sound a little weird as it may be interpreted as the final state of the task is "done". "markAsNotDone" could be a better alternative.

Reply:
I like the alternative that you suggest. The "unmarkAsDone" has changed to "markAsNotDone".